.vsts.ci.yml

```diff
@@ -0,0 +1,163 @@
+- phase: build_linux_agent
+################################################################################
+  displayName: Linux Agent
+  queue:
+    demands: agent.os -equals linux
+  steps:
+
+  # Clean
+  - checkout: self
+    clean: true
+
+  - script: ./dev.sh layout
+    workingDirectory: src
+    displayName: Build & Layout
+
+  - script: ./dev.sh test
+    workingDirectory: src
+    displayName: Test
+    
+  - script: ./dev.sh layout Release
+    workingDirectory: src
+    displayName: Build & Layout Release
+    condition: and(succeeded(), ne(variables['build.reason'], 'PullRequest'))
+
+  - script: ./dev.sh package Release
+    workingDirectory: src
+    displayName: Package Release
+    condition: and(succeeded(), ne(variables['build.reason'], 'PullRequest'))
+
+  - task: PublishBuildArtifacts@1
+    displayName: Publish Artifact (Linux)
+    condition: and(succeeded(), ne(variables['build.reason'], 'PullRequest'))
+    inputs:
+      pathToPublish: _package
+      artifactName: agent
+      artifactType: container
+
+################################################################################
+- phase: build_osx_agent
+################################################################################
+  displayName: OSX Agent
+  queue:
+    demands: agent.os -equals darwin
+  steps:
+
+  # Clean
+  - checkout: self
+    clean: true
+
+  - script: ./dev.sh layout
+    workingDirectory: src
+    displayName: Build & Layout
+
+  - script: ./dev.sh test
+    workingDirectory: src
+    displayName: Test
+    
+  - script: ./dev.sh layout Release
+    workingDirectory: src
+    displayName: Build & Layout Release
+    condition: and(succeeded(), ne(variables['build.reason'], 'PullRequest'))
+
+  - script: ./dev.sh package Release
+    workingDirectory: src
+    displayName: Package Release
+    condition: and(succeeded(), ne(variables['build.reason'], 'PullRequest'))
+
+  - task: PublishBuildArtifacts@1
+    displayName: Publish Artifact (OSX)
+    condition: and(succeeded(), ne(variables['build.reason'], 'PullRequest'))
+    inputs:
+      pathToPublish: _package
+      artifactName: agent
+      artifactType: container
+
+################################################################################
+- phase: publish_agent_packages
+################################################################################
+  displayName: Publish Agents (Windows/Linux/OSX)
+  queue:
+    demands: agent.os -equals Windows_NT
+  dependsOn: 
+  - build_windows_agent
+  - build_osx_agent
+  - build_linux_agent
+  #condition: and(succeeded(), eq(variables.publish, true))
+  condition: and(succeeded(), ne(variables['build.reason'], 'PullRequest'))
+  steps:
+  
+  - checkout: none
+  
+  - task: DownloadBuildArtifacts@0
+    displayName: Download Agent Packages
+    condition: and(succeeded(), ne(variables['build.reason'], 'PullRequest'))
+    inputs:
+      artifactName: agent
+
+  - powershell: |
+      Import-Module AzureRM
```

Comment:
Auto-resolving should work (although explicit import is better). If you do explicit import, I think sometimes the top level AzureRM module might not be there. You might want to do:
```powershell
Write-Host "Preloading Azure modules." # This is for better performance, to avoid module-autoloading.
Import-Module AzureRM, AzureRM.profile, AzureRM.Storage, Azure.Storage, AzureRM.Cdn -ErrorAction Ignore -PassThru
```

You might want to verify -PassThru is supported on powershell v4.